lib/chef-dk/policyfile/chef_server_cookbook_source.rb

```diff
@@ -15,28 +15,41 @@
 # limitations under the License.
 #
 
+require 'ffi_yajl'
 require 'chef-dk/exceptions'
+require 'chef-dk/policyfile/source_uri'
+require 'chef/http/simple'
 
 module ChefDK
   module Policyfile
     class ChefServerCookbookSource
-
       attr_reader :uri
 
       def initialize(uri)
-        @uri = uri
+        @uri = SourceURI.parse(uri)
+        @http_connections = {}
+        yield self if block_given?
       end
 
       def ==(other)
         other.kind_of?(self.class) && other.uri == uri
       end
 
       def universe_graph
-        raise UnsupportedFeature, 'ChefDK does not support chef-server cookbook default sources at this time'
+        @universe_graph ||= begin
+          full_chef_server_graph.inject({}) do |normalized_graph, (cookbook_name, metadata_by_version)|
+            normalized_graph[cookbook_name] = metadata_by_version.inject({}) do |deps_by_version, (version, metadata)|
+              deps_by_version[version] = metadata["dependencies"]
+              deps_by_version
+            end
+            normalized_graph
+          end
+        end
       end
 
       def source_options_for(cookbook_name, cookbook_version)
-        raise UnsupportedFeature, 'ChefDK does not support chef-server cookbook default sources at this time'
+        base_uri = full_chef_server_graph[cookbook_name][cookbook_version]['download_url']
+        { artifactserver: base_uri, version: cookbook_version }
```

Comment:
let's add `:http_client` as a source option so that the client for the relevant cookbook can be dependency injected into the OmniFetcher?